scipy/stats/_morestats.py

```diff
@@ -1288,6 +1304,28 @@ def _all(x):
         message = ("The `optimizer` argument of `boxcox_normmax` must return "
                    "an object containing the optimal `lmbda` in attribute `x`.")
         raise ValueError(message)
+    else:
+        # Test if the optimal lambda causes overflow
+        max_x = np.max(x, axis=0)
+        istransinf = np.isinf(special.boxcox(max_x, res))
+        dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
+        if np.any(istransinf):
+            warnings.warn(
+                f"The optimal lambda is {res}, but the returned lambda is"
+                " the constrained optimum to ensure that maximum of"
+                f" transformed data does not cause overflow in {dtype}."
```

Comment:
```suggestion
                f"The optimal lambda is {res}, but the returned lambda is "
                f"the constrained optimum to ensure that maximum of " 
                f"transformed data does not cause overflow in {dtype}."
```

Reply:
We don't, it is purely a style preference for me. If you think it looks better without the `f`, feel free to not include it, but I think it looks better to have the lines aligned.